### PR TITLE
[Woo POS] Remove custom text style. Extract overlay into reusable base in product info dialog

### LINKED ISSUE #12322: [Woo POS] M2: Remove custom text style, extract overlay into reusable base in product info dialog
The goal of this task is to remove custom text style and instead use the one available from `MaterialTheme` and extract overlay used in product info dialog into reusable base for all dialogs.

Context: https://github.com/woocommerce/woocommerce-android/pull/12267#pullrequestreview-2235199513

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12322

### Description

The PR extracts reusable component for the dialogs. Adds some adjustments to the layouts. Removes usage of the custom fonts

### Testing information

Make sure there is not regression with POS exit dialog and dialog about simple products

### The tests that have been performed

I tested all known to me cases of the usages of the dialogs. Check the video

### Images/gif

https://github.com/user-attachments/assets/7858b41b-54a0-48f3-b3af-0e4c3fa40cee

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosExitConfirmationDialog.kt [hunk 1]
```diff
@@ -45,119 +29,71 @@ import com.woocommerce.android.ui.woopos.common.composeui.toAdaptivePadding
...
+            .fillMaxWidth()
+            .padding(148.dp.toAdaptivePadding()),
+        isVisible = isVisible,
+        dialogBackgroundContentDescription = stringResource(
+            id = R.string.woopos_dialog_exit_confirmation_background_content_description
+        ),
+        onDismissRequest = onDismissRequest,
     ) {
-        Box(
+        Card(
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosExitConfirmationDialog.kt]: ❓ Do we need `Card` component here? `WooPosDialogWrapper` already wraps the content composable in a `Card` component.

**Review by AnirudhBhat — APPROVED** [review]
Thanks for the improvement @kidinov. LGTM! I've just left one minor comment that is non-blocking for the merge.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosExitConfirmationDialog.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosExitConfirmationDialog.kt]: Good catch @AnirudhBhat!

I removed that f30316f522a0ee91afd410b0a311226d1ef47847

**Pull request merged**